pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/ComparisonColumns.java

```diff
@@ -78,18 +69,10 @@ public int compareTo(ComparisonColumns other) {
     // _comparisonColumns should only at most one non-null comparison value for newly ingested data. If not, it is
     // the user's responsibility. There is no attempt to guarantee behavior in the case where there are multiple
     // non-null values
-    int comparisonResult;
-
     Comparable comparisonValue = _values[_comparableIndex];
     Comparable otherComparisonValue = other.getValues()[_comparableIndex];
 
-    if (otherComparisonValue == null) {
-      // Keep this record because the existing record has no value for the same comparison column, therefore the
-      // (lack of) existing value could not possibly cause the new value to be rejected.
-      comparisonResult = 1;
-    } else {
-      comparisonResult = comparisonValue.compareTo(otherComparisonValue);
-    }
+    int comparisonResult = comparisonValue.compareTo(otherComparisonValue);
```

Comment:
no more null-checking required, because _values will always hold all non-null entries (where some may be `<defaultNullValue>`